sql/src/main/java/org/apache/druid/sql/calcite/planner/PlannerContext.java

```diff
@@ -502,4 +507,30 @@ public void setJoinExpressionVirtualColumnRegistry(VirtualColumnRegistry joinExp
   {
     this.joinExpressionVirtualColumnRegistry = joinExpressionVirtualColumnRegistry;
   }
+
+  public String getStatementKind()
+  {
+    return this.statementKind;
+  }
+
+  public void setStatementKind(String sqlKind)
+  {
+    if (this.statementKind != null) {
+      throw new ISE("StatementKind has already been set");
+    }
+    this.statementKind = sqlKind;
+  }
+
+  public SqlNode getTargetDataSource()
```

Comment:
Should we annotate this with `@Nullable`?

Reply:
Done